src/ZipkinGuzzle/Middleware.php

```diff
@@ -61,15 +62,20 @@ function (ResponseInterface $response) use ($span, $scopeCloser) {
                     if ($response->getStatusCode() > 399) {
                         $span->tag(Tags\ERROR, true);
                     }
+
                     $span->finish();
                     $scopeCloser();
                     return $response;
                 },
                 function ($reason) use ($span, $scopeCloser) {
-                    $response = $reason instanceof RequestException
-                        ? $reason->getResponse()
-                        : null;
-                    $span->tag(Tags\ERROR, true);
+                    $error = true;
+                    $response = null;
+                    if ($reason instanceof RequestException) {
+                        $response = $reason->getResponse();
+                        $error = $reason->getMessage();
```

Comment:
Previously the error was `true` (because I still don't know what $reason could be if not `RequestException`).